src/utils/queries.js

```diff
@@ -7,4 +7,13 @@ const getResource = async (_key, id) => {
   return data;
 };
 
-export { getResource };
+const getResources = async searchTerm => {
+  const url =
+    searchTerm !== ''
+      ? `${API_URL}/resources/?search=${searchTerm}` // Empty '' search term will return 0 results
```

Comment:
It makes sense that an empty search term will return 0 results, but it's a bit of an over reach to query the `/resources/` endpoint instead, which would return _all_ or a paginated set of resources IIRC, right? If we're doing this, then the conditional in line 46 of the resource/index page will never get hit 
https://github.com/codebuddies/frontend/pull/162/files#diff-cec2f6afae4583cc82d24feda5d79ad6R46

Reply:
When the component is initially loaded, I have `searchValue` initialized to equal `''` an empty string but if I search for an empty string, I'll get 0 results: 

![image](https://user-images.githubusercontent.com/4512699/94352931-53370400-0020-11eb-9a4b-6f02103f2496.png)

However, if I type to search `''` explicitly in the searchbox or a search term that isn't found, I'll get the expected 0 results:
![image](https://user-images.githubusercontent.com/4512699/94352940-69dd5b00-0020-11eb-82ce-cddc0eaa02ee.png)
![image](https://user-images.githubusercontent.com/4512699/94352946-79f53a80-0020-11eb-84e8-1022dde78bec.png)

I'm not sure if this answers your question? Was a bit confused by the question, since the conditional in line 46 in `resources/index.js` does get hit from what I understand...

Reply:
I think I'm mainly finding it odd the way that we're validating the searchTerm in the query method itself. Maybe we should be validating in the frontend before it gets passed to here?

Reply:
Hmm... how would we validate in the frontend?